### ISSUE TITLE
Refine regexes in default excludes to support coverage for monorepos in "alle" structure

### ISSUE THREAD

**Issue by amb26** [istanbuljs/istanbuljs#146]
A popular alternative monorepo structure to the "lerna" structure is described in the "alle" proposal at https://gist.github.com/nolanlawson/457cdb309c9ec5b39f0d420266a9faa4 . This is adopted by several projects including PouchDB which is a highly popular long-running project with hundreds of contributors. The rationale behind the layout is explained in the gist - the PouchDB team worked with the "lerna" structure for some years before deciding it didn't work for them.

The current implementation at https://github.com/istanbuljs/istanbuljs/blob/master/packages/test-exclude/index.js#L35 is hard-wired to add the following pattern to default excludes if there are not any:

    **/node_modules/** 

This excludes all files in which "node_modules" appears anywhere in their path, rather than just those which have been deposited into the package by an npm run.

A possibility for refining the regexes might be to instead use

    node_modules/**

but there may be other options.

The best workaround we've been able to think of for this problem is rather messy and time-consuming and can be seen at https://github.com/the-t-in-rtf/universal/blob/e0609d04b2c6074fe9142f8f239bb08052d3acea/.nycrc - the strategy is to forcibly include every file individually that we require coverage for by means of a negated exclude, e.g.

```
  "exclude": [
    "!**/gpii/node_modules/accessRequester/index.js",
    "!**/gpii/node_modules/accessRequester/src/accessRequester.js",
    "!**/gpii/node_modules/canopyMatchMaker/index.js",
...
```

Any suggestions for an improved workaround would be welcome.

**Comment by coreyfarrell** [comment]
We've added an option `excludeNodeModules` which you can set to `false` to prevent addition of `**/node_modules/**`.  For your situations you would probably need to add `node_modules/**` to your exclude option.  This functionality was released in nyc 14.0.0.

**Comment by the-t-in-rtf** [comment]
Great, thanks!
